src/middlewared/middlewared/plugins/device_/device_info_linux.py

```diff
@@ -192,10 +192,15 @@ def get_disk_details(self, block_device, disk, disks_data):
 
             # get lunid
             if disk_data['wwn']:
-                if disk.get('tran') == 'nvme':
+                if disk_data['tran'] == 'nvme':
                     disk['lunid'] = disk_data['wwn'].lstrip('eui.')
                 else:
                     disk['lunid'] = disk_data['wwn'].lstrip('0x')
+            # USB disks can use the GUID from the partition table as a uniqe ID.
+            if disk_data['tran'] == 'usb' and disk_data['pttype'] == 'gpt':
```

Comment:
That will only work if the disk is partitioned.

What if you add 4 new unformatted disks to your USB enclosure?

Reply:
This is a intentional, since TrueNAS doesn't have a physical disk adoption process compared to other more standard OS distributions. You have to prep disks with GPT ahead of time.

Reply:
We don't think this is a reasonable solution.

GPT header gets wiped as soon as a pool is created with the disks.

How about we use the long serial if its USB?